networkx/algorithms/tree/junction_tree_algorithm.py

```diff
@@ -0,0 +1,81 @@
+r"""Function for computing a junction tree of a graph."""
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+from networkx.algorithms import moral, complete_to_chordal_graph, chordal_graph_cliques
+from itertools import combinations
+
+__all__ = ["junction_tree"]
+
+
+@not_implemented_for("multigraph", "MultiDiGraph")
+def junction_tree(G):
+    r"""Returns a junction tree of a given graph.
+
+    Parameters
+    ----------
+    G : networkx.Graph
+        Directed graph
+
+    Returns
+    -------
+    junction_tree : networkx.Graph
+        The corresponding junction tree of `G`.
+
+    Raises
+    ------
+    NetworkXNotImplemented
+        Raised if `G` is an instance of `MultiGraph` or `MultiDiGraph`.
+
+    Notes
+    -----
+    A junction tree (or clique tree) is a tree generated from a (un)directed graph G.
+    The tree's nodes consist of cliques and connecting sepsets of the moralized and
+    triangulated version of G respectively. The sepset of two cliques is the intersection
+    of the variables of these cliques, e.g. the sepset of (A,B,C) and (A,C,E,F) is (A,C).
+    The junction tree algorithm consists of five steps [1]_:
+
+    1. Moralize graph
+    2. Triangulate graph
+    3. Find maximum cliques
+    4. Build graph from cliques, connecting cliques with shared
+       variables, set edge-weight to number of shared variables
+    5. Find maximum spanning tree
+
+    References
+    ----------
+    .. [1] Junction tree algorithm:
+       https://en.wikipedia.org/wiki/Junction_tree_algorithm
+
+    .. [2] Finn V. Jensen and Frank Jensen. 1994. Optimal
+       junction trees. In Proceedings of the Tenth international
+       conference on Uncertainty in artificial intelligence (UAI’94).
+       Morgan Kaufmann Publishers Inc., San Francisco, CA, USA, 360–366.
+    """
+
+    clique_graph = nx.Graph()
+    if G.is_directed():
+        moralized = moral.moral_graph(G)
+        chordal_graph, _ = complete_to_chordal_graph(moralized)
+    else:
+        chordal_graph, _ = complete_to_chordal_graph(G)
```

Comment:
```suggestion
    if G.is_directed():
        G = moral.moral_graph(G)
    chordal_graph, _ = complete_to_chordal_graph(G)
```

Reply:
I'm glad you mentioned the naming! I had a similar idea a few days ago but somehow forgot about it. I choose "decomposition", which makes sense :) I also implemented your suggested change.